pkg/azure/ipam/node.go

```diff
@@ -173,8 +173,43 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 		}
 		return nil
 	})
+	if err != nil {
+		return nil, -1, err
+	}
+
+	requiredIfaceName := n.k8sObj.Spec.Azure.InterfaceName
+	err = n.manager.instances.ForeachInterface(n.node.InstanceID(), func(instanceID, interfaceID string, interfaceObj ipamTypes.InterfaceRevision) error {
+		iface, ok := interfaceObj.Resource.(*types.AzureInterface)
+		if !ok {
+			return fmt.Errorf("invalid interface object")
+		}
+
+		if requiredIfaceName != "" {
+			if iface.Name != requiredIfaceName {
+				scopedLog.WithFields(logrus.Fields{
+					"ifaceName":    iface.Name,
+					"requiredName": requiredIfaceName,
+				}).Debug("Not considering interface is available since it does not match the required name")
+				return nil
+			}
+		}
+
+		scopedLog.WithFields(logrus.Fields{
+			"id":           iface.ID,
+			"numAddresses": len(iface.Addresses),
+		}).Debug("Considering interface is available")
+
+		availableOnInterface := math.IntMax(types.InterfaceAddressLimit-len(iface.Addresses), 0)
+		if availableOnInterface > 0 {
+			remainAvailableENIsCount++
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, -1, err
+	}
```

Comment:
Similarly to the comment above, let's move this into a separate function. Otherwise the logic will diverge over time.

Reply:
Is mean error  handle need into a separate function? I move available logic in separate function.
``` go
availableOnInterface, available := isAvailableInterface(requiredIfaceName, iface, scopedLog)
		if !available {
			return nil
		}
```